tools/Google.Cloud.Tools.GenerateDocfxSources/Program.cs

```diff
@@ -203,11 +203,45 @@ private static string TransformDocTemplate(ApiMetadata api, string text)
 The credentials will automatically be used to authenticate. See the [Getting Started With
 Authentication](https://cloud.google.com/docs/authentication/getting-started) guide for more details.";
 
+            string clientClasses = CreateClientClassesDocumentation(api);
+
+            string clientConstruction =
+@"Create a client instance by calling the static `Create` method, optionally
+specifying an end-point or channel and settings";
+
             return text
                 .Replace("{{title}}", title)
                 .Replace("{{description}}", description)
                 .Replace("{{installation}}", installation)
-                .Replace("{{auth}}", auth);
+                .Replace("{{auth}}", auth)
+                .Replace("{{client-classes}}", clientClasses)
+                .Replace("{{client-construction}}", clientConstruction);
+        }
+
+        private static string CreateClientClassesDocumentation(ApiMetadata api)
+        {
+            if (api.Type != "grpc")
+            {
+                return "FIXME"; // No automatic templating for this API
+            }
+            var layout = DirectoryLayout.ForApi(api.Id);
+            var packageSource = Path.Combine(layout.SourceDirectory, api.Id);
+            var sourceFiles = Directory.GetFiles(packageSource, "*Client.cs");
+            var clients = sourceFiles
+                .Where(file => File.ReadAllText(file).Contains(": ServiceSettingsBase")) // Check it contains a generated client
+                .Select(file => Path.GetFileName(file))             // Just the file name, not full path
+                .Select(file => file.Substring(0, file.Length - 3)) // Trim .cs
+                .OrderBy(client => client)
+                .Select(client => $"[{client}](obj/api/{api.Id}.{client}.yml)") // Markdown link to API doc
+                .ToList();
+            switch (clients.Count)
+            {
+                case 0: return "FIXME"; // No automatic templating for this API
+                case 1: return $"All operations are performed through {clients[0]}.";
+                default:
+                    var list = string.Join("\r\n", clients.Select(client => $"- {client}"));
```

Comment:
Is "\r\n" definitely required? Rather than `Environment.NewLine`?

Reply:
I wondered about that. Ultimately, the rest of the text has line breaks of "however it was checked out from GitHub" which is kinda hard to replicate. I thought that this would at least be consistent and obvious. I suspect it won't matter anyway and git will canonicalize one way or another :)